docs/dcp2_system_design.rst

```diff
@@ -0,0 +1,1465 @@
+with the metadata that describes them, and subgraphs that link them to
+their input sequence files. Azul indexes these subgraphs and stitches
+them to the primary subgraphs containing the sequence files and
+experimental metadata. For each project, the Azul indexer compiles a
+stratified tree of DCP/2-generated matrices, in addition to the tree of
+`Contributor-generated matrices (CGMs)`_. The Data
+Browser displays the DCP/2 matrices in the files tab and on the project
+details page, similar to how it displays `Contributor-generated matrices
+(CGMs)`_.
+
+Like CGMs, DCP/2-generated matrices are stratified. While the
+stratification of a CGM is supplied out-of-band, by hijacking a property
+of a ``supplementary_file`` entity, the stratification of a
+DCP/2-generated matrix is determined organically by traversing the
+metadata graph.
+
+The DCP/2 analysis team assembles stratified project-level matrices from
+smaller intermediate matrices. The project-level matrix will be staged
+in a separate subgraph that links the project-level matrix to the
+intermediate matrices. The subgraphs containing the intermediate
+matrices link to the input sequence files which are in separate primary
+subgraphs. Azul recursively stitches these three levels of subgraphs
+together.
+
+Project level matrix files are described by an ``analysis_file`` entity
+with the following properties:
+
+-  The ``analysis_file.provenance.submitter_id`` field is set to
+   ``e67aaabe-93ea-564a-aa66-31bc0857b707``. This UUIDv5 was
+   generated using the same namespace UUID as for
+   `Contributor-generated matrices (CGMs)`_ and the name
+   ``dcp2``.
+
+-  The ``analysis_file.file_core.content_description`` field of both
+   intermediate and project-level matrices is set to::
+
+       {
+           "text": "DCP/2-generated matrix",
+           "ontology": "data:3917",
+           "ontology_label": "Count Matrix"
+       }
+
+-  The ``analysis_file`` entities are linked to inputs via
+       ``analysis_process`` entities.
+
+-  Each ``analysis_process`` is described by an ``analysis_protocol``
+
+-  The ``analysis_protocol.protocol_core.protocol_id`` will be
+       ``optimus_post_processing_v1.0.0`` for stratified, project-level
+       matrices and ``optimus_v1.0.0`` for the intermediate ones
+
+Any intermediate matrices created during the processing are described as
+``analysis_file``, but the ``analysis_file.provenance.submitter_id``
+property is omitted.
+
+.. figure:: media/dcp2-matrices.png
+
+   Graphical interpretation of how DCP/2 generated matrices are
+   linked to the experimental graph
+
+DCP/1-generated matrices
+------------------------
+
+-  The ``supplementary_file.provenance.submitter_id`` field is set to
+       ``c9efbb15-c50c-5796-8d15-35e9e1219dc5``. This UUIDv5 was
+       generated using the same namespace UUID as for
+       `Contributor-generated matrices (CGMs)`_ and the name
+       ``dcp1 matrix service``.
```

Comment:
```suggestion
-  The ``supplementary_file.provenance.submitter_id`` field is set to ``c9efbb15-c50c-5796-8d15-35e9e1219dc5``. This UUIDv5 was generated using the same namespace UUID as for `Contributor-generated matrices (CGMs)`_ and the name ``dcp1 matrix service``.
```
Line wrapping and indentation in source causes odd bolding of first line and indentation in displayed output.